pyquil/external/rpcq.py

```diff
@@ -1,56 +1,163 @@
+import json
 from typing import Dict, List, Union, Optional, Any
-from typing_extensions import Literal
-from pydantic import BaseModel, Field
 from rpcq.messages import TargetDevice as TargetQuantumProcessor
+from dataclasses import dataclass, field
 
+#JsonValue = Union[type(None), bool, int, float, str, list["JsonValue"], dict[str, "JsonValue"]]
+JsonValue = Any
 
-class Operator(BaseModel):
+@dataclass
+class Operator:
     operator: Optional[str] = None
     duration: Optional[float] = None
     fidelity: Optional[float] = None
 
-
+    def __post_init__(self):
+        self.duration = float(self.duration) if self.duration is not None else None
+        self.fidelity = float(self.fidelity) if self.fidelity is not None else None
+
+    def to_dict(self) -> Dict[str, JsonValue]:
+        if type(self) is Operator:
+            raise ValueError("Should be a subclass")
+        return dict(
+            operator=self.operator,
+            duration=self.duration,
+            fidelity=self.fidelity,
+        )
+
+    @classmethod
+    def from_dict(cls, dictionary):
+        operator_type = dictionary["operator_type"]
+        if operator_type == "measure":
+            return MeasureInfo.from_dict(dictionary)
+        if operator_type == "gate":
+            return GateInfo.from_dict(dictionary)
+        raise ValueError("Should be a subclass")
+
+
+@dataclass
 class MeasureInfo(Operator):
     qubit: Optional[Union[int, str]] = None
     target: Optional[Union[int, str]] = None
-    operator_type: Literal["measure"] = "measure"
 
+    def __post_init__(self):
+        self.qubit = str(self.qubit)
 
-class GateInfo(Operator):
-    parameters: List[Union[float, str]] = Field(default_factory=list)
-    arguments: List[Union[int, str]] = Field(default_factory=list)
-    operator_type: Literal["gate"] = "gate"
+    def to_dict(self) -> Dict[str, JsonValue]:
+        encoding = super().to_dict()
+        encoding["operator_type"] = "measure"
+        encoding["qubit"] = self.qubit
+        encoding["target"] = self.target
+        return encoding
 
+    @classmethod
+    def from_dict(cls, dictionary: dict) -> "MeasureInfo":
+        return MeasureInfo(
+            operator=dictionary.get("operator"),
+            duration=dictionary.get("duration"),
+            fidelity=dictionary.get("fidelity"),
+            qubit=dictionary.get("qubit"),
+            target=dictionary.get("target"),
+        )
 
-class Qubit(BaseModel):
+
+@dataclass
+class GateInfo(Operator):
+    parameters: List[Union[float, str]] = field(default_factory=list)
+    arguments: List[Union[int, str]] = field(default_factory=list)
```

Comment:
It was a field with type `Literal["gate"]` included in the BaseModel just so that it would appear in the json.  Since I'm writing my own `to_json` method here, it seemed unnecessary to keep the field.  But I don't mind putting it back if there is a reason to.

Reply:
I think we should keep it for backwards compatibility.

Reply:
Restored!